virtcontainers/sandbox.go

```diff
@@ -1156,13 +1167,23 @@ func (s *Sandbox) StartContainer(containerID string) (VCContainer, error) {
 }
 
 // StopContainer stops a container in the sandbox
-func (s *Sandbox) StopContainer(containerID string) (VCContainer, error) {
+func (s *Sandbox) StopContainer(containerID string) (_ VCContainer, err error) {
 	// Fetch the container.
 	c, err := s.findContainer(containerID)
 	if err != nil {
 		return nil, err
 	}
 
+	defer func() {
```

Comment:
Is it better to move this check into c.stop()?  Another question, when the qemu process exited by accident, then this stop would take a long time to wait grpc time out, can we do this check before call agent.stopContainer() too, if the hypervisor process exited, then skip this agent calling, this will save some time. The same to stopVM(). What do you think @bergwolf @sboeuf ?

Reply:
Yes, make sense, I know the question,  `when the qemu process exited by accident, then this stop would take a long time to wait grpc time out,`, but not find a good way. 

I find one more problem in the patch, we need to unmount host mount and rootfs , they are in `c.sandbox.agent.stopContainer` function, if `RemoveContainerRequest` fails, it will skip the following step